generate_directory_indexes.py

```diff
@@ -15,37 +22,68 @@
-
-logger = logging.getLogger(__name__)
-if args.verbose == None:
-    logging.basicConfig(level=logging.ERROR)
-if args.verbose == 1:
-    logging.basicConfig(level=logging.WARNING)
-elif args.verbose == 2:
-    logging.basicConfig(level=logging.INFO)
-else:
-    logging.basicConfig(level=logging.DEBUG)
+script = os.path.basename(sys.argv[0])
+configuration = None
+
+def parse_arguments():
+    global configuration
+    global logger
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "path",
+        help="Top directory for writing index files.")
+    parser.add_argument(
+        "--file-metadata", "-f",
+        default=None,
+        help="A file containing data describing the tree to index.")
+    parser.add_argument(
+        "--metadata-delimiter", "-m",
+        default=";",
+        help="Character which seperates fields in the file metadata.")
+    parser.add_argument(
+        "--noop", "-n",
+        help="Only print files to be created without writing them to disk",
+        action="store_true")
+    parser.add_argument(
+        "--verbose", "-v",
+        help="Verbose output. Repeat (up to -vvv) for more verbosity",
+        action="count")
+    configuration = parser.parse_args()
+
+    logger = logging.getLogger(__name__)
+    logger_format = '%(filename)s: %(levelname)s %(message)s'
+
+    if configuration.verbose == 1:
+        logging.basicConfig(level=logging.WARNING, format=logger_format)
+    elif configuration.verbose == 2:
+        logging.basicConfig(level=logging.INFO, format=logger_format)
+    elif configuration.verbose > 2:
+        logging.basicConfig(level=logging.DEBUG, format=logger_format)
+    else:
+        logging.basicConfig(level=logging.ERROR, format=logger_format)
+
 
 def index_link(prefix, current_order_by, new_order_by, reverse_order):
     if current_order_by == new_order_by:
         return index_file_name(prefix, current_order_by, not reverse_order)
     else:
         return index_file_name(prefix, new_order_by, False)
 
-# use a templating library to turn a prefix and a list of contents into an HTML directory index
+####
+# use a templating library to turn a prefix and a list of contents
+# into an HTML directory index
+
 def render_index(prefix, order_by, contents, reverse_order, base_path):
     logger.debug('rendering index for {prefix} ordered by {order_by} and reverse_order={reverse_order}'.format(prefix=prefix, order_by=order_by, reverse_order=reverse_order))
 
 
     sorted_contents = sorted(contents, key=lambda k: k[order_by], reverse=reverse_order)
-    formatted_contents = format_directory_listing(sorted_contents)
+    formatted_contents = format_file_details(sorted_contents)
+
+    # Remove the base path from the prefix to avoid putting the full
+    # filesystem path in the index
 
-    # Remove the base path from the prefix to avoid putting the full filesystem path in the index
     path = '' if prefix == base_path else prefix.replace(base_path, '')
```

Comment:
I know this wasn't part of your changeset, but couldn't this just be `path = prefix.replace(base_path, '')`?

Reply:
Quite possibly. I'd have to tinker with it to see if that excludes some edge case that isn't obvious. Left alone for now.